### PR TITLE
Upgrade calico to 3.27.4

### PR DESCRIPTION
**Pull request by addyess**
### Overview

Upgrade calico version deployed with the charm

### Details
* stop downloading the full 1GB release file, just the calicoctl versions necessary
* updating the manifest files after syncing the images to rocks.cc
* changing the update.py script to make it easier to push using:

```bash
tox -e update -- --registry upload.rocks.canonical.com:5000/cdk
```

### CODE REVIEW

File: build-calico-resource.sh [hunk 1]
```diff
@@ -47,16 +47,13 @@ function fetch_and_validate() {
   fi
 }
 
-wget \
-  https://github.com/projectcalico/calico/releases/download/$calico_version/release-$calico_version.tgz
-tar -xf release-$calico_version.tgz
-
```
Comment (addyess) [inline build-calico-resource.sh:53]: This file is REALLY big --  over 1G... inside it are two files we use and LOTS we don't

File: build-calico-resource.sh [hunk 2]
```diff
@@ -47,16 +47,13 @@ function fetch_and_validate() {
...
 for arch in ${arches}; do
   rm -rf resource-build-$arch
   mkdir resource-build-$arch
   pushd resource-build-$arch
-  cp ../release-$calico_version/bin/calicoctl/calicoctl-linux-$arch calicoctl
-
+  fetch_and_validate \
+    https://github.com/projectcalico/calico/releases/download/$calico_version/calicoctl-linux-$arch
+  mv calicoctl-linux-$arch calicoctl
```
Comment (addyess) [inline build-calico-resource.sh:56]: rather than copy those two files from the extracted tarball -- just download the calicoctl that we need for that arch

**Review by addyess — COMMENTED** [review]
Comments about the build resources change

**Review by mateoflorido — APPROVED** [review]
LGTM

**Pull request merged**